### PR TITLE
Add pgvector datastore

### PR DESCRIPTION
**Pull request by sahusiddharth**
fixes: #142

### CODE REVIEW

**Comment by sahusiddharth** [comment]
Hey @anakin87,

I'm diving into implementing vector data store for the first time, and I have made some progress. However, there are a few areas where I could use your expertise:

### Configuration File (.toml):
I've made some headway on the feature, but I'm stuck on finalizing the .toml file. I couldn't fully grasp what was happening there. Any guidance you can provide would be immensely helpful.

### Test Case Implementation:
I haven't had much experience writing test cases before, and I want to ensure robust testing for this feature. Your assistance in creating effective test cases would be invaluable to ensure the reliability of the codebase.

### Retriever Module (retriever.py):
In the `run` function of the `retriever.py` module, there are some aspects I'm unsure about, and I have a few questions. I believe your insights could help me iron out the details.

I appreciate your support.

**Comment by anakin87** [comment]
Hello, @sahusiddharth.

I took the time to get a first look...
This integration won't be trivial but can be a good experience.

### Client
- As we also discussed offline, it is better to use the official [pgvector python client](https://github.com/pgvector/pgvector-python)
- I would also suggest using the [SQLAlchemy implementation](https://github.com/pgvector/pgvector-python?tab=readme-ov-file#sqlalchemy) but this is not mandatory

### Other suggestions
- I strongly suggest having a local installation of pgvector and also testing any change/implementation on your machine
- https://github.com/deepset-ai/haystack-core-integrations/issues/142#issuecomment-1875451116

### What I will do
- I will setup for you the GitHub testing workflow, so for each commit, the CI will run tests that you can use to drive your development process (of course, you can still run them locally)

Feel free to ask specific questions/doubts...

File: integrations/pgvector/tests/test_document_store.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+class TestDocumentStore(CountDocumentsTest, DeleteDocumentsTest, WriteDocumentsTest):
+
+    @pytest.fixture
+    def docstore(self) -> PGvectorDocumentStore:
+        """
+        This is the most basic requirement for the child class: provide
+        an instance of this document store so the base class can use it.
+        """
+        return PGvectorDocumentStore() # FIXME
```
Comment (anakin87) [inline integrations/pgvector/tests/test_document_store.py:23]: @sahusiddharth here you should put a valid instance of your Document Store to run tests...
Currently, it is only a placeholder.

**Comment by anakin87** [comment]
I added the GitHub workflow that runs tests. Now they are running. See also https://github.com/deepset-ai/haystack-core-integrations/pull/172#discussion_r1443038616

**Comment by sahusiddharth** [comment]
On it

**Comment by sahusiddharth** [comment]
Thank you @anakin87

**Comment by anakin87** [comment]
Hey, @sahusiddharth...
If you need help, feel free to let me know!

**Comment by sahusiddharth** [comment]
@anakin87 

Thanks for checking on me.
Was out of station, didn't get a change to resume working on it, I'll start today

**Comment by anakin87** [comment]
No rush! If you need any help or clarification, please ask...

**Comment by anakin87** [comment]
Since there was no progress on this PR for a while, we decided to implement the `PgvectorDocumentStore` in another PR (#239).

@sahusiddharth thank you for your efforts.
I see that you have already found other opportunities to contribute to Haystack. Thank you again!

**Pull request closed without merging**